LFS-647: Time question type

acrowthe opened the pull request:
Add a time question type using the browser's built in time picker
- Add support for this type in the questionnaire builder

The [test branch](https://github.com/ccmbioinfo/lfs/tree/LFS-647-test) has 4 examples of the time question in the demographics form, and more questions can be added or edited via the questionnaire editor in either branch.

Known issue: The Google Chrome time picker on linux may not allow for hour values higher than 19, despite a lack of AM/PM selector. This can be worked around using the time picker UI ([LFS-657](https://phenotips.atlassian.net/browse/LFS-657))

acrowthe committed:
LFS-647: Time question type
Add a time question type using the browser's built in time picker
- Add support for this type in the wuestionnaire builder

marta- commented on modules/data-entry/src/main/frontend/src/questionnaire/TimeQuestion.jsx:
```diff
@@ -0,0 +1,164 @@
+      && this.minutes < 60 && this.minutes >= 0
+  }
+
+  numberToDoubleDigit(num) {
```
Another option is to use `` `${num}`.padStart(2, "0") ``

acrowthe committed:
LFS-647: Time question type
Improve number padding

marta- reviewed (approved):
Looks good!

sashaandjic reviewed (approved):
tested

sdumitriu commented on modules/data-entry/src/main/frontend/src/questionnaire/TimeQuestion.jsx:
```diff
@@ -0,0 +1,160 @@
+
+  let checkError = (timeString) => {
+    let time = new Time(timeString);
+    if ((upperLimit && upperTime < time) || (lowerLimit && lowerTime >= time)) {
```
Should time's own valid status be checked as well?

sdumitriu replied:
If `minValue` is `10:00`, shouldn't `10:00` be accepted? I would expect it to be.

```suggestion
    if ((upperLimit && upperTime < time) || (lowerLimit && lowerTime > time)) {
```

sdumitriu reviewed (changes requested):
When invalid values are saved and the form is loaded again, the error messages don't show up. Can that be fixed?

Backend validation is missing as well, but that's a bigger issue to be addressed later.

acrowthe committed:
LFS-647: Time question type
Improve error checking and handling
- Check if a valid time is provided if an answer is required
- Check answer validity on loading the page
- Fix lower limit to be inclusive
